pkg/kubectl/cmd/util/helpers.go

```diff
@@ -83,21 +83,28 @@ func DefaultBehaviorOnFatal() {
 	fatalErrHandler = fatal
 }
 
-// fatal prints the message and then exits. If V(2) or greater, glog.Fatal
-// is invoked for extended information.
+// fatal prints the message (if provided) and then exits. If V(2) or greater,
+// glog.Fatal is invoked for extended information.
 func fatal(msg string) {
-	// add newline if needed
-	if !strings.HasSuffix(msg, "\n") {
-		msg += "\n"
-	}
-
 	if glog.V(2) {
 		glog.FatalDepth(2, msg)
```

Comment:
Fatal doesn't require a trailing newline (it strips it).

On May 12, 2016, at 12:58 PM, Janet Kuo <notifications@github.com> wrote:

In pkg/kubectl/cmd/util/helpers.go
<https://github.com/kubernetes/kubernetes/pull/25451#discussion_r63056896>:

>  func fatal(msg string) {
> -	// add newline if needed
> -	if !strings.HasSuffix(msg, "\n") {
> -		msg += "\n"
> -	}
> -
>  	if glog.V(2) {
>  		glog.FatalDepth(2, msg)

Should "add newline" be before this?

—
You are receiving this because you were mentioned.
Reply to this email directly or view it on GitHub
<https://github.com/kubernetes/kubernetes/pull/25451/files/4fd51ec8308924b241ad852bf9b32220cfa5874d#r63056896>